cyf-ecommerce-api/server.js

```diff
@@ -0,0 +1,364 @@
+app.post("/customers/:customerId/orders", (req, res) => {
+  const orderDate = req.body.order_date;
+  const orderReference = req.body.order_reference;
+  const customerId = Number(req.params.customerId);
+
+  if (!orderDate) {
+    return res.status(400).json({
+      result: "failure",
+      msg: "An orderDate value is required.",
+    });
+  } else if (!orderReference) {
+    return res.status(400).json({
+      result: "failure",
+      msg: "An orderReference value is required.",
+    });
+  }
+
+  pool
+    .query(
+      `SELECT * FROM customers INNER JOIN orders ON orders.customer_id = customers.id WHERE customers.id = $1`,
+      [customerId]
+    )
+    .then((result) => {
+      if (result.rows.length == 0) {
+        return res.status(400).json({
+          result: "failure",
+          msg: `Invalid customerId value.`,
+        });
+      } else {
+        const query = `INSERT INTO orders (order_date, order_reference, customer_id) VALUES($1, $2, $3)`;
+        pool
+          .query(query, [orderDate, orderReference, customerId])
+          .then(() =>
+            res.json({
+              msg: "Order has been successful created.",
+            })
+          )
+          .catch((error) => {
+            res.status(400).json(error);
+          });
+      }
+    })
+    .catch((error) => res.status(400).json(error));
+});
+
+//- Add a new PUT endpoint `/customers/:customerId` to update an existing customer (name, address, city and country).
+app.put("/customers/:customerId", (req, res) => {
+  const customerId = Number(req.params.customerId);
+  console.log(customerId);
+
+  const customerName = req.body.name;
+  const customerAddress = req.body.address;
+  const customerCity = req.body.city;
+  const customerCountry = req.body.country;
+
+  pool
+    .query(`SELECT * FROM customers WHERE id = $1`, [customerId])
+    .then((result) => {
+      if (result.rows.length == 0) {
+        return res.status(400).json({
+          result: "failure",
+          msg: `Invalid customerId value.`,
+        });
+      } else {
+        const query = `UPDATE customers SET name =$1 , address = $2, city = $3, country = $4  WHERE id= $5`;
+        pool
+          .query(query, [
+            customerName,
+            customerAddress,
+            customerCity,
+            customerCountry,
+            customerId,
+          ])
+          .then(() =>
+            res.json({
+              msg: `The customer's details/detail with id ${customerId} have/has been successful updated`,
+            })
+          )
+          .catch((error) => {
+            res.status(401).json(error);
+          });
+      }
+    })
+    .catch((error) => res.status(400).json(error));
+});
+
+//- Add a new DELETE endpoint `/orders/:orderId` to delete an existing order along with all the associated order items.
+app.delete("/orders/:orderId", function (req, res) {
+  const orderId = req.params.orderId;
+
```

Comment:
remember to check the validity of the orderId before you query (you can use the checks done before for length and is a number here before querying) => if this was not valid then you return the message: Invalid orderId